tests/test_core.py

```diff
@@ -9193,6 +9193,15 @@ def test_pthread_dylink_longjmp(self):
     main = test_file('core/pthread/test_pthread_dylink_longjmp.c')
     self.dylink_testf(main, need_reverse=False)
 
+  @needs_dylink
+  @node_pthreads
+  def test_pthread_dylink_main_module_1(self):
+    self.emcc_args.append('-Wno-experimental')
+    self.set_setting('EXIT_RUNTIME')
+    self.set_setting('USE_PTHREADS')
+    self.set_setting('MAIN_MODULE')
+    self.do_runf(test_file('hello_world.c'))
```

Comment:
Is this test related?

Reply:
Yes, this test fails before this change since with `__call_main` being undefined.